Sources/EdgeState.swift

```diff
@@ -33,21 +33,20 @@ class EdgeState {
     ///
     /// - Parameters:
     ///   - event: The `Event` triggering the bootup
-    ///   - getXDMSharedState: used to fetch the Consent data
-    func bootupIfNeeded(event: Event, getXDMSharedState: @escaping (_ name: String, _ event: Event?, _ barrier: Bool) -> SharedStateResult?) {
+    ///   - getSharedState: used to fetch the `Event Hub` shared state
+    func bootupIfNeeded(event: Event, getSharedState: @escaping (_ name: String, _ event: Event?, _ barrier: Bool) -> SharedStateResult?) {
         guard !hasBooted else { return }
 
         // check if consent extension is registered
-        if let consentSharedState = getXDMSharedState(EdgeConstants.SharedState.Consent.SHARED_OWNER_NAME, nil, false) {
-            currentCollectConsent = ConsentStatus.getCollectConsentOrDefault(eventData: consentSharedState.value ?? [:])
+        if let registeredExtensionsWithHub = getSharedState(EdgeConstants.SharedState.Hub.SHARED_OWNER_NAME, event, false)?.value,
+           let extensions = registeredExtensionsWithHub[EdgeConstants.EventDataKeys.EXTENSIONS] as? [String: Any],
+           extensions[EdgeConstants.SharedState.Consent.SHARED_OWNER_NAME] as? [String: Any] != nil {
+            // keep consent pending until the consent preferences update event is received
```

Comment:
This expression seems unconventional to me, the only content on the true case is a comment. 
It'd make more sense to see it written as something like to avoid the empty condition case.

```swift
if !consentRegistered {
   updateCurrentConsent(status: EdgeConstants.Defaults.COLLECT_CONSENT_YES)
}